go/libkb/sig_chain.go

```diff
@@ -385,41 +410,49 @@ var hardcodedResets = map[keybase1.SigID]bool{
 // GetCurrentSubchain takes the given sigchain and walks backward until it
 // finds the start of the current subchain, returning all the links in the
 // subchain. See isSubchainStart for the details of the logic here.
-func (sc *SigChain) GetCurrentSubchain(eldest keybase1.KID) ([]*ChainLink, error) {
+func (sc *SigChain) GetCurrentSubchain(eldest keybase1.KID) (ChainLinks, error) {
+	return backtrackUntilNextEldest(sc.chainLinks, eldest)
+}
+
+// backtrackUntilNextEldest takes the given set of chain links, and then limits the tail
+// of the chain to just those that correspond to the eldest key given by `eldest`.
+func backtrackUntilNextEldest(links []*ChainLink, eldest keybase1.KID) (ChainLinks, error) {
```

Comment:
Below I wonder if we could drop the `eldest` param here and lift that part of the logic into one (but not both) of the callers. If we do that, maybe we could call it just `getRightmostSubchain` or something?